pkg/ddl/ddl_worker.go

```diff
@@ -1398,68 +1397,12 @@ func waitSchemaChanged(d *ddlCtx, waitTime time.Duration, latestSchemaVersion in
 	return checkAllVersions(d, job, latestSchemaVersion, timeStart)
 }
 
-func checkAllVersions(d *ddlCtx, job *model.Job, latestSchemaVersion int64, timeStart time.Time) error {
-	failpoint.Inject("checkDownBeforeUpdateGlobalVersion", func(val failpoint.Value) {
-		if val.(bool) {
-			if mockDDLErrOnce > 0 && mockDDLErrOnce != latestSchemaVersion {
-				panic("check down before update global version failed")
-			}
-			mockDDLErrOnce = -1
-		}
-	})
-
-	// OwnerCheckAllVersions returns only when all TiDB schemas are synced(exclude the isolated TiDB).
-	err := d.schemaSyncer.OwnerCheckAllVersions(d.ctx, job.ID, latestSchemaVersion)
-	if err != nil {
-		logutil.Logger(d.ctx).Info("wait latest schema version encounter error", zap.String("category", "ddl"), zap.Int64("ver", latestSchemaVersion),
-			zap.Int64("jobID", job.ID), zap.Duration("take time", time.Since(timeStart)), zap.Error(err))
-		return err
-	}
-	logutil.Logger(d.ctx).Info("wait latest schema version changed(get the metadata lock if tidb_enable_metadata_lock is true)", zap.String("category", "ddl"),
-		zap.Int64("ver", latestSchemaVersion),
-		zap.Duration("take time", time.Since(timeStart)),
-		zap.String("job", job.String()))
-	return nil
-}
-
 // waitSchemaSyncedForMDL likes waitSchemaSynced, but it waits for getting the metadata lock of the latest version of this DDL.
 func waitSchemaSyncedForMDL(d *ddlCtx, job *model.Job, latestSchemaVersion int64) error {
```

Comment:
Do we need to move this function too？

Reply:
I prefer to MDL related code into one file like `mdl.go`